File: 1_Review/python/1_basic_concepts.py

# printing to the console
# print is built-in python function
print("It's me")

# must define custom methods before you can use them

# create a function that returns None
def greet():
    print("Hello, World!")
    print("Hello again")

def greet2():
    print("Hello, World!: ", end="")

# creates a function that returns a number
def add(num1, num2):
    return num1 + num2

# greet function does not execute until you call it
greet()
greet()

greet2()
greet2()

# prints 3 to the console
print(add(1,2))

# prints None to the console
print(greet())


items = [1,2,3]
items2 = items * 2
print(items2)

items3 = [item * 2 for item in items]
print(items3)

items4 = map(lambda item: item * 2, items)
print(items4)

items5 = list(items4)
print(items5)

File: 1_Review/csharp/1_basic_concepts.cs
// Printing to the console.
Console.WriteLine("It's me");

// Custom methods must be declared before they are called.
static void Greet()
{
    Console.WriteLine("Hello, World!");
    Console.WriteLine("Hello again");
}

static void Greet2()
{
    Console.Write("Hello, World!: ");
}

static int Add(int number1, int number2)
{
    return number1 + number2;
}

// A method does not execute until it is called.
Greet();
Greet();

Greet2();
Greet2();
Console.WriteLine();

// Prints 3 to the console.
Console.WriteLine(Add(1, 2));

// A void method has no return value.
Greet();

var items = new List<int> { 1, 2, 3 };
var items2 = items.Concat(items).ToList();
Console.WriteLine(string.Join(", ", items2));

var items3 = items.Select(item => item * 2).ToList();
Console.WriteLine(string.Join(", ", items3));

var items4 = items.Select(item => item * 2);
// The following line will print the type of items4, not its individual elements
Console.WriteLine(items4);

var items5 = items4.ToList();
Console.WriteLine(string.Join(", ", items5));